src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringInterpolation.cs

```diff
@@ -401,6 +501,162 @@ private static void AssertNoImplicitInterpolatedStringHandlerConversions(Immutab
+                    partCall.Method.Name == BoundInterpolatedString.AppendLiteralMethod
+                    ? partCall.Arguments[0]
+                    : new BoundStringInsert(partCall.Syntax, partCall.Arguments[0], null, null, false)
+                );
+            }
+            return builder.ToImmutableAndFree();
+        }
+
+        private (ImmutableArray<BoundExpression> Calls, int IncreasedLiteralLength, int FilledHolesCount) OptimizeAppendFormattedCalls(ImmutableArray<BoundExpression> parts)
+        {
+            var result = ArrayBuilder<BoundExpression>.GetInstance();
+            var currentStringConst = PooledStringBuilder.GetInstance();
+            (BoundCall? AppendLiteralCall, BoundLiteral? StringLiteral) reusable = default;
+            BoundExpression? lastAppendReceiver = null;
+            int filledHolesCount = 0;
+            int increasedLiteralLength = 0;
+
+            BoundExpression createAppendLiteralCallFromBoundLiteral(BoundLiteral literal, BoundExpression receiver) =>
+                _factory.InstanceCall(receiver, BoundInterpolatedString.AppendLiteralMethod, literal);
+            BoundExpression createAppendLiteralCallFromString(string literal, BoundExpression receiver) =>
+                createAppendLiteralCallFromBoundLiteral(_factory.StringLiteral(literal), receiver);
+            void mergeReusableIntoBuilder()
+            {
+                if (reusable.StringLiteral.ConstantValueOpt != null)
+                {
+                    currentStringConst.Builder.Append(reusable.StringLiteral.ConstantValueOpt.StringValue);
+                }
+                reusable = default;
+            }
+            void mergeReusableIntoResult(BoundCall? call = null)
+            {
+                switch (reusable)
+                {
+                    case (not null, _):
+                        result.Add(reusable.AppendLiteralCall);
+                        break;
+                    case (_, not null):
+                        Debug.Assert(call?.ReceiverOpt is not null || lastAppendReceiver is not null);
+                        result.Add(createAppendLiteralCallFromBoundLiteral(reusable.StringLiteral, call?.ReceiverOpt ?? lastAppendReceiver!));
+                        break;
+                    default:
+                        if (currentStringConst.Length > 0)
+                        {
+                            Debug.Assert(call?.ReceiverOpt is not null || lastAppendReceiver is not null);
+                            result.Add(createAppendLiteralCallFromString(currentStringConst.ToStringAndFree(), call?.ReceiverOpt ?? lastAppendReceiver!));
+                            currentStringConst = PooledStringBuilder.GetInstance();
+                        }
+                        return;
+                }
+                reusable = default;
+            }
+
+            foreach (var part in parts)
+            {
+                if (part is not BoundCall { Method.Name: not null, } call)
```

Comment:
When is this true? What does it mean?

Reply:
I assume `BoundDynamicCall`.
It should be treated as is.

Reply:
I mean `BoundDynanmicInvoke`.